### ISSUE TITLE
Add a knowledge base

### ISSUE THREAD

**Issue by HelenWDTK** [mysociety/alaveteli#7764]
We should develop an on site knowledge base to help address common support issues and user/authority queries. This should be easily accessible and searchable, enabling users to self-serve by finding the information that is relevant to them, and administrators to quickly resolve support queries by responding with a link. Over time, this feature may supersede current help pages. By integrating the help system into the app from its current position within the theme, we would make it quicker and easier for site administrators to update and maintain.

The proposed knowledge base should seamlessly integrate with the current tagging system.

I see this working much like the existing category system.  Help entries linked to a specific tag should be consolidated on a single page, replicating how requests and authorities under the same tag are displayed. Multiple tags per entry could allow the same entry to be featured across different pages. By mirroring the category system, we could construct subpages for common help topics, which would add much needed structure to our advice and support.

Each entry on a page should be individually collapsible and linkable, reducing the length of current help pages and simplifying the process of pointing to specific advice.

Building a tag based knowledge base may potentially enable us to display associated help information across different locations in the app, possibly through extracting the contents into a note or similar.

In terms of the back end, I think the interface for adding new entries should be accessible from the existing tagging page next to requests, blog posts, outgoing and incoming messages etc. I would expect the page where I added a new entry to look much like the existing form that we use to add tag based notes. This would need to capture the heading for the entry, and ideally would include a way to include basic formatting. I would also expect to see an admin tag that is like the existing category creation page, that would enable admins to associate tags with "topics" in the knowledge base, in the same way that we associate public body tags with categories now.

For clarity, imagine this is a giant FAQ. Each "question" is tangible and linkable, and the tags determine which "questions" are displayed where.

Related: https://github.com/mysociety/alaveteli/issues/233

**Comment by HelenWDTK** [comment]
Example of a knowledge base that displays collapsable help, allows searching and has categories of topics - ours would be better: https://help.sainsburys.co.uk/help

**Comment by WilliamWDTK** [comment]
This is definitely a good idea, I would say.

---

One could pose the question as to whether this is worth the time implementing it directly into Alaveteli, or whether there is an existing (open-source?) solution that could be integrated so as to offer an appropriate interface for less developer time.

Such a solution might work for Alavetelis run by organisation who might also wish to have a knowledgebase for other services they run, where such tagging could separate by service.

I do not know if such an exact system exists.

However, if it is envisage to tightly integrate it into Alaveteli's functionality, one could easily see how it would be better to implement it directly.

**Comment by HelenWDTK** [comment]
In my mind, adding/editing entries would look a bit like this
![IMG_20230607_162320080](https://github.com/mysociety/alaveteli/assets/120410992/7c6a0a9d-cbda-469c-b8bb-201df5d9a359)
Adding structure would look like the existing category creation/ editing workflow linking tags to topics/sub topics:
<img width="677" alt="Screenshot 2023-06-07 at 16 33 42" src="https://github.com/mysociety/alaveteli/assets/120410992/92322fe3-b7e1-4c0d-8de6-693aa6792c58">

**Comment by garethrees** [comment]
We currently have some links around the site that link to a specific page and anchor (`/help/privacy#name`), which is universal across Alaveteli installs. We'd have to figure out how we preserve this behaviour, or do something different such that something similar still works for reusers.

**Comment by HelenWDTK** [comment]
I think there would still be a place for a small number of static pages with the standard boilerplate legal stuff on - eg Privacy Policy. That would not preclude adding information about those topics to the knowledge base.

**Comment by garethrees** [comment]
We could add a slug attribute (editable, but auto-generated from the title) that maps to the on-page html anchor, and include a list of required section/anchors that we match against the current database records to check for missing pieces; something like:

```rb
class KnowledgeBaseController
  cattr_accessor :required_sections, default: %w[
    privacy#name
    requesting#missing_body
    …
  ]
end
```

Could then display that on the admin interface as a bit of a checklist for things to write about.

![Screenshot 2023-06-12 at 16 25 01](https://github.com/mysociety/alaveteli/assets/282788/52a90617-b96e-44c7-8564-978f51880184)

Each required section could have an associated partial (e.g. `app/views/admin/knowledge_base/required_sections/privacy/_name.html`) that describes what it should be about; maybe with some example phrasing, but allows the site admins to write text that's appropriate to the local install.

**Comment by garethrees** [comment]
> Adding structure would look like the existing category creation/ editing workflow linking tags to topics/sub topics:

Currently there's no manual ordering options for the pulled in content (only the headings/subheadings). That would probably be pretty tricky to build. Does this still work okay, even if we can't order at the entries level?
